### PR TITLE
Buffered transactional editing

### LINKED ISSUE #7387: Buffered Transactional Editing (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#46750
Author: @domi4484
QGIS version: 3.26 (Feature)

**Buffered Transactional Editing**

## Buffered Transactional Editing

Implementation of [QGIS-Enhancement-Proposals #203](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/203)
This pr includes the main implementation and follow #46483

Brief summary:
A new "buffered transactions" editing mode for QGIS is added.
With this edit mode, all editable layers are toggled synchronously and all edits are saved in a local edit buffer.
Saving changes is executed within a single transaction on all layers (per provider).

image

In the gif below layers from different providers are simultaneously put into edit mode and persisted.

bufferedTransactions

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by domi4484**
Documentation for QGIS feature introduced here: [https://github.com/qgis/QGIS/pull/46750](https://github.com/qgis/QGIS/pull/46750)
Documentation issue Fixes #7387

### CODE REVIEW

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 1]
```diff
@@ -1484,15 +1484,23 @@ Data Sources Properties
...
-* |unchecked| :guilabel:`Automatically create transaction groups where possible`:
-  When this mode is turned on, all
-  layers from the same database are synchronised in their edit state,
-  i.e. when one layer is put into edit state, all are, when one layer is committed
-  or one layer is rolled back, so are the others. Also, instead of buffering edit
-  changes locally, they are directly sent to a transaction in the database which
-  gets committed when the user clicks save layer.
-  Note that you can (de)activate this option only if no layer is being edited
-  in the project.
+* :guilabel:`Transaction mode`:
```
Comment (DelazJ) [inline docs/user_manual/introduction/qgis_configuration.rst:1487]: Would you have a one-line sentence to describe this entry?

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -1484,15 +1484,23 @@ Data Sources Properties
...
+
+  * :guilabel:`Local Edit Buffer` Edits are buffered locally and sent to the provider 
+    when toggling layer editing mode.
+  * :guilabel:`Automatic Transaction Groups` Automatic transactional editing means 
+    that on supported datasources (postgres and geopackage databases) the edit state
```
Comment (DelazJ) [inline docs/user_manual/introduction/qgis_configuration.rst:1492]: Colon looks nicer and avoids repeating.
```suggestion
  * :guilabel:`Automatic Transaction Groups`:
    on supported datasources (postgres and geopackage databases) the edit state 
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -1484,15 +1484,23 @@ Data Sources Properties
...
+  * :guilabel:`Local Edit Buffer` Edits are buffered locally and sent to the provider 
+    when toggling layer editing mode.
+  * :guilabel:`Automatic Transaction Groups` Automatic transactional editing means 
+    that on supported datasources (postgres and geopackage databases) the edit state 
+    of all tables that originate from the same database are synchronized and executed 
+    in a server side transaction. Also, instead of buffering edit changes locally, they 
+    are directly sent to a transaction in the database which gets committed when the user 
+    clicks save layer.
+  * :guilabel:`Buffered Transaction Groups` Buffered transactional editing means that 
+    all editable layers in the buffered transaction group are toggled synchronously and
```
Comment (DelazJ) [inline docs/user_manual/introduction/qgis_configuration.rst:1498]: Here we use `Buffered Transaction Groups` to explain the same expression. Confusing IMHO. Would this work (copied from the previous option)?
```suggestion
  * :guilabel:`Buffered Transaction Groups`:
    all editable layers that originate from the same database are toggled synchronously and 
```

**Review by DelazJ — COMMENTED** [review]
@domi4484 Thanks. I have some rephrasing suggestions and questions.

File: docs/user_manual/introduction/qgis_configuration.rst (hunk 1 above)
Comment (domi4484) [reply docs/user_manual/introduction/qgis_configuration.rst:1487]: Maybe something like "Define how edits are sent to the data provider"?

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 4]
```diff
@@ -1484,15 +1484,23 @@ Data Sources Properties
...
-  i.e. when one layer is put into edit state, all are, when one layer is committed
-  or one layer is rolled back, so are the others. Also, instead of buffering edit
-  changes locally, they are directly sent to a transaction in the database which
-  gets committed when the user clicks save layer.
-  Note that you can (de)activate this option only if no layer is being edited
-  in the project.
+* :guilabel:`Transaction mode`:
+
+  * :guilabel:`Local Edit Buffer` Edits are buffered locally and sent to the provider 
+    when toggling layer editing mode.
```
[the remark this answers is not in the document]
Comment (domi4484) [reply docs/user_manual/introduction/qgis_configuration.rst]: Yes, also when pressing "save edits", I have changed this and also reformulated it the same way for the automatic transaction group option

File: docs/user_manual/introduction/qgis_configuration.rst (hunk 3 above)
Comment (domi4484) [reply docs/user_manual/introduction/qgis_configuration.rst:1498]:
```suggestion
  * :guilabel:`Buffered Transaction Groups`:
    all editable layers are toggled synchronously and 
```
In fact all editable layer are saved/put in editing mode at the same time, it doesn't matter from which database/provider

**Comment by domi4484** [comment]
@DelazJ thank you for the review! I have added some comments and modifications

**Comment by DelazJ** [comment]
@domi4484 I'm fine with your suggestions. thanks

File: docs/user_manual/introduction/qgis_configuration.rst (hunk 1 above)
Comment (domi4484) [reply docs/user_manual/introduction/qgis_configuration.rst:1487]:
```suggestion
* :guilabel:`Transaction mode`: define how edits are sent to the data provider```

**Comment by DelazJ** [comment]
one last thing I just notice: can you update the screenshot, please?

**Comment by domi4484** [comment]
I haven't found exactly the same project as the old screenshot (missing 2 raster layers). The size is also not the exactly the same. @DelazJ Is that ok or should i try to get more close to the old screenshot?

**Comment by DelazJ** [comment]
Yes, it's OK. I can't tell you the one used myself so no worries.

**Pull request merged**